src/platform/windows.rs

```diff
@@ -171,3 +173,30 @@ impl DeviceIdExtWindows for DeviceId {
         self.0.persistent_identifier()
     }
 }
+
+/// Additional methods on `Icon` that are specific to Windows.
+pub trait IconExtWindows: Sized {
+    /// Create an icon from a file path.
+    ///
+    /// Specify `size` to load a specific size from the file, or `None` to load the default
+    /// icon size from the file.
+    fn from_path<P: AsRef<Path>>(path: P, size: Option<(usize, usize)>) -> Result<Self, BadIcon>;
```

Comment:
```suggestion
    /// Create an icon from a file path.
    ///
    /// Specify `size` to load a specific icon size from the file, or `None` to load the default
    /// icon size from the file.
    fn from_path<P: AsRef<Path>>(path: P, size: Option<PhysicalSize<u32>>) -> Result<Self, BadIcon>;
```

Also, what happens if the icon file doesn't contain the specified size?

Reply:
From some quick experiments, it looks like windows will attempt to scale the icon if the desired size isn't present.